modules/custom/activity_basics/activity_basics.module

```diff
@@ -12,6 +12,22 @@ use Drupal\node\NodeInterface;
  * Implements hook_entity_insert().
  */
 function activity_basics_entity_insert(EntityInterface $entity) {
+  switch ($entity->getEntityTypeId()) {
```

Comment:
@OH17L, I didn't test it yet, but I do have a question.

I like the idea of doing this before we even create an activity, but I do have a few concerns.

You now have two cases (comment and vote). How about private messages, nodes or other entities (that may not yet exist)? And what about digest (daily or weekly) emails. They will collect a bunch of activities and will simply still send that (last) one out.

Reply:
@bramtenhove, you are right. I will extend my functionality.

Reply:
@OH17L but I'm wondering if this is maintainable in the long run. It could be easier to just check before sending the email (two places) if the user is active or not.

See https://github.com/goalgorilla/open_social/blob/8.x-2.x/modules/custom/activity_send/modules/activity_send_email/src/Plugin/EmailFrequency/Immediately.php and https://github.com/goalgorilla/open_social/blob/8.x-2.x/modules/custom/activity_send/modules/activity_send_email/src/Plugin/QueueWorker/ActivityDigestWorker.php

Reply:
@bramtenhove, when I delete my code here and add conditions in two files where you propose then a user will receive all e-mails which were sent when this user was blocked. Is it ok?

Reply:
@OH17L sorry, I'm not sure I understand. You still get emails or not?

Reply:
@bramtenhove, no, I do not get emails.

Reply:
@OH17L ok! 👍

Then I suggest we just implement the change in those two files. Agreed?

Reply:
@bramtenhove, yes, I agree and I will do that.